src/stdlib_stringlist.f90

```diff
@@ -0,0 +1,996 @@
+    class(stringlist_type), intent(inout)        :: list
+    type(stringlist_index_type), intent(in)      :: idx
+    character(len=*), intent(in)                 :: string
+
+    integer                                      :: idxabs
+
+    idxabs = merge( idx%offset, list%size + idx%offset, idx%head )
+
+    call list%insert( idxabs, string )
+end subroutine insert_string_idx_wrap
+
+subroutine insert_stringlist_idx_wrap( list, idx, slist )
+    class(stringlist_type), intent(inout)        :: list
+    type(stringlist_index_type), intent(in)      :: idx
+    class(stringlist_type), intent(in)           :: slist
+
+    integer                                      :: idxabs
+
+    idxabs = merge( idx%offset, list%size + idx%offset, idx%head )
+
+    call list%insert( idxabs, slist )
+end subroutine insert_stringlist_idx_wrap
+
+subroutine insert_stringarray_idx_wrap( list, idx, sarray )
+    class(stringlist_type), intent(inout)        :: list
+    type(stringlist_index_type), intent(in)      :: idx
+    character(len=*), dimension(:), intent(in)   :: sarray
+
+    integer                                      :: idxabs
+
+    idxabs = merge( idx%offset, list%size + idx%offset, idx%head )
+
+    call list%insert( idxabs, sarray )
+end subroutine insert_stringarray_idx_wrap
+
+! insert_empty_positions
+!     Insert a number of positions for new strings
+!
+! Arguments:
+!     list                   The list of strings where the empty positions should be inserted
+!     idxn                   Index at which the positions should be inserted
+!     number                 Number of positions
+!
+subroutine insert_empty_positions( list, idxn, number )
+    class(stringlist_type), intent(inout)        :: list
+    integer, intent(inout)                       :: idxn
+    integer, intent(in)                          :: number
+
+    integer                                      :: i, inew
+    integer                                      :: lastidx
+    type(string_type), dimension(:), allocatable :: new_string
+
+    !
+    ! Clip the index between 1 and size+1
+    !
+    idxn = max( 1, min(list%size+1, idxn ) )
+
+    !
+    ! Check if the array list%string is large enough
+    ! Make room in any case
+    !
+    if ( .not. allocated(list%string) ) then
+        allocate(list%string(1) )
+    endif
+
+    lastidx = list%size + number
+
+    !
+    ! Do we need a copy?
+    !
+    if ( size(list%string) < lastidx ) then
+        allocate( new_string(lastidx) )
+
+        do i = 1,idxn-1
+            call move_alloc( list%string(i)%value, new_string(i)%value )
+        enddo
+
+        do i = idxn, list%size
+            inew = i + number
+            call move_alloc( list%string(i)%value, new_string(inew)%value )
+        enddo
+        call move_alloc( new_string, list%string )
+    else
+        do i = idxn, list%size
+            inew = i + number
+            call move_alloc( list%string(i)%value, list%string(inew)%value )
```

Comment:
`move_alloc(FROM, TO)`
@arjenmarkus if I have understood it correctly what happens if there is already a string at index `inew` (`TO`)?
If it gets overwritten we will lose what was stored at index `inew`.

for example:
`list = ["1", "2", "3", "4", "5", "6", "7", "8"]`
and I do something like this:
```fortran
do i = 3, 8
    inew = i + 3
    call move_alloc(list%string(i)%value , list%string(inew)%value)
```
Then I will get 
`["1", "2", _, _, _, _, _, _, "3", "4", "5"]`
`_` represents that it is deallocated.